app/models/comment.rb

```diff
@@ -64,6 +75,12 @@ def initialize(person, target, text)
       super(person, target)
     end
 
+    def build(*args)
+      super.tap do |comment|
+        comment.filter_mentions(comment.people_allowed_to_be_mentioned)
```

Comment:
> Not really. Have a look on this test -- it passes well, that means that mentions work on private receive.

That test is not really useful:
1. you don't know who can see this post on receive, because you only receive the first visibility.
2. my local test (where I saw that it doesn't work with private) was: podA  has your code, user@podB mentions user@podC (which is also a receiver of this post), but the mention was filtered (because podA doesn't know about that).

You can filter local users and recreate filtered mentions on second receive, if the second user can see it now, but I doubt that this is practicable.

So that all is no real solution for the problem. I think the most reasonable solution is to filter the streams, so you can only view posts, that you are allowed to see. We should still filter on outgoing stuff (because old posts would otherwise show the posts to users who shouldn't see it), but incoming private posts are 
impossible to filter, because you don't know who can see it.

But my original intention with this topic (filter on receive) was, to ensure that a user don't receives a notification for a mention in a comment, if he can't see the post. If this is done on notification-level (like for notifications for mentions in posts, where this already works), I'm totally fine with it.